public-ui/src/app/views/ehr-view/ehr-view.component.ts

```diff
@@ -45,29 +45,41 @@ export class EhrViewComponent implements OnInit, OnDestroy {
   synonymString = {};
   /* Show different graphs depending on domain we are in */
   graphToShow = GraphType.BiologicalSex;
-  showTopConcepts = false;
+  showTopConcepts: boolean;
   medlinePlusLink: string;
   graphButtons = [];
 
   @ViewChild('chartElement') chartEl: ElementRef;
 
 
   constructor(private route: ActivatedRoute,
+    private router: Router,
     private api: DataBrowserService,
-    private tooltipText: TooltipService
+    private tooltipText: TooltipService,
   ) {
     this.route.params.subscribe(params => {
       this.domainId = params.id;
     });
+    console.log(this.router.onSameUrlNavigation);
   }
   ngOnInit() {
+    this.loadPage();
```

Comment:
moved inint logic to loadPage().
loadPage() can be called instead of ngOnInit() to refresh view